src/polysolve/LinearSolverCuSolverDN.cu

```diff
@@ -0,0 +1,161 @@
+#ifdef POLYSOLVE_WITH_CUSOLVER
+
+////////////////////////////////////////////////////////////////////////////////
+#include <polysolve/LinearSolverCuSolverDN.cuh>
+#include <fstream>
+#include <string>
+#include <iostream>
+
+#include "spdlog/spdlog.h"
+
+#define gpuErrchk(ans){ gpuAssert((ans), __FILE__, __LINE__); }
```

Comment:
this is bad, do you need it?

Reply:
I thought it was common practice to check GPU code for errors by putting each CUDA code line in a function like this

Reply:
macros are bad, we can have a function

Reply:
made it a function